<?xml version="1.0"?>
<launch>
  <arg name="simulation" default="false" />
  <arg name="uav_odom_topic_name" default="uav_odom" />
  <arg name="image_name" default="/camera/image_rect" />
  
  <!-- vision base -->
  <include file="$(find jsk_uav_forest_perception)/launch/image_process.launch">
    <arg name="independent_nodelet" value="$(arg simulation)" />
    <arg name="image_name" value="$(arg image_name)" unless="$(arg simulation)"/>
  </include>

  <!-- 2D laser base -->
  <include file="$(find jsk_uav_forest_perception)/launch/laser_process.launch">
  </include>

  <!-- tree detector -->
  <node pkg="jsk_uav_forest_perception" type="tree_detector_node" output="screen" name="tree_detector_node">
    <param name="color_region_center_topic_name" value="/object_image_center" />
    <param name="camera_info_topic_name" value="/camera/camera_info" />
    <param name="laser_scan_topic_name" value="/shadow_filtered" />
    <param name="clurstering_max_radius" value="0.4" />
    <param name="uav_odom_topic_name" value="/ground_truth/state" if="$(arg simulation)"/>
    <param name="uav_odom_topic_name" value="$(arg uav_odom_topic_name)" unless="$(arg simulation)"/>
    <param name="verbose" value="false" />
  </node>

</launch>
